src/Image/Darkroom/ImageMagick.php

```diff
@@ -72,8 +72,47 @@ protected function convert(string $file, array $options): string
 			}
 		}
 
+		// frame option to allow selecting layers for multi-layer or frames for animated images
+		$fileOptions = '';
+		$frame = $options['frame'];
+		$maxFrames = $this->frameCount($file, $options);
+		if ($frame !== null) {
+			// check whether frame is in bounds
+			if ($frame < 0) {
+				throw new Exception('Frame option must be a positive integer');
+			}
+
+			if ($frame !== 0 && $frame >= $maxFrames) {
+				throw new Exception('Frame option must be smaller than the number of frames in the image');
+			}
+
+			$fileOptions = "[{$frame}]";
+		} elseif ($maxFrames > 1) {
+			// if frame is not set and target format doesn't support multi-layer images, select first frame
+			$targetFormat = $options['format'] ?? F::extension($file);
+			$multiLayerFormats = ['gif', 'avif', 'webp'];
+			if (!in_array($targetFormat, $multiLayerFormats)) {
+				$fileOptions = '[0]';
+			}
+		}
+
 		// append input file
-		return $command . ' ' . escapeshellarg($file);
+		return $command . ' ' . escapeshellarg($file . $fileOptions);
+	}
+
+	/**
+	 * Returns the number of frames in an image
+	 */
+	public function frameCount(string $file, array $options): int
+	{
+		exec($options['identifyBin'] . ' -ping -format "%n\n" ' . escapeshellarg($file), $output, $return);
+
+		// log broken commands
+		if ($return !== 0) {
+			throw new Exception('The ImageMagick frame identification command could not be executed: ' . $file);
```

Comment:
Yes. 

While ImageMagick 7+ includes both commands by default as 'identify' and 'convert', they might not be available on the server under the default name, in which case the error would throw.

Reply:
But wondering how widespread that is already (I am not much familiar with it, sorry) - just concerned that suddenly many of our IM users would run into this exception

Reply:
It should be available everywhere where `convert` is installed, as it comes with the default bundle of ImageMagick.

Reply:
When trying to fix the conflicts, I ran into this breaking out unit test setup. I am wondering if it would be better to fall back to use the first frame here instead of throwing an exception.